### PR TITLE
Migrate two party POL native tests to local interchain

### PR DESCRIPTION
**Pull request by keyleu**

### CODE REVIEW

File: local-interchaintest/src/utils/constants.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+pub const GAIA_CHAIN: &str = "gaia";
+pub const GAIA_CHAIN_ID: &str = "localcosmos-1";
+
+pub const OSMOSIS_CHAIN: &str = "osmosis";
+pub const OSMOSIS_CHAIN_ID: &str = "localosmosis-1";
+
+pub const STRIDE_CHAIN: &str = "stride";
+pub const STRIDE_CHAIN_ID: &str = "localstride-3";
+
+pub const CHAIN_CONFIG_PATH: &str = "local-interchaintest/chains/neutron_gaia_osmosis_stride.json";
```
Comment (bekauz) [inline local-interchaintest/src/utils/constants.rs:16]: let's add a note in the readme that tests should be executed from the root of the repo given how the paths are configured

**Review by bekauz — COMMENTED** [review]
this is looking great.

couple of minor comments about the structure and steps to running the tests locally. will revisit the actual test soon.

File: local-interchaintest/src/utils/constants.rs (hunk 1 above)
Comment (keyleu) [reply local-interchaintest/src/utils/constants.rs:16]: Added.

File: local-interchaintest/src/helpers/astroport.rs
[the remark this answers is not in the document]
Comment (keyleu) [reply local-interchaintest/src/helpers/astroport.rs:1]: Yeah I'll probably refactor this once we also have Osmosis, which I think is the next one I'll migrate.

File: local-interchaintest/src/utils/setup.rs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            if local_chain.contract_codes.contains_key(id) {
+                info!(
+                    "Contract {} already deployed on chain {}, skipping...",
+                    id, chain
+                );
+                continue;
+            }
+
+            let code_id = cw.store(ACC_0_KEY, abs_path.as_path()).unwrap();
+
```
[the remark this answers is not in the document]
Comment (keyleu) [reply local-interchaintest/src/utils/setup.rs:68]: Added a cache which stores in a local json file after first execution.

File: local-interchaintest/src/utils/setup.rs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+use log::info;
+
+use super::{
+    constants::{ACC_0_KEY, WASM_EXTENSION},
+    file_system::read_artifacts,
+    test_context::TestContext,
+};
+
+pub fn deploy_contracts_on_chain(test_ctx: &mut TestContext, path: &str, chain: &str) {
+    let dir_entries = read_artifacts(path).unwrap();
```
[the remark this answers is not in the document]
Comment (keyleu) [reply local-interchaintest/src/utils/setup.rs:26]: Added a check there.

File: local-interchaintest/src/utils/liquid_staking.rs
[the remark this answers is not in the document]
Comment (keyleu) [reply local-interchaintest/src/utils/liquid_staking.rs:1]: Yes, for now I didn't add anything for stride liquid staking yet but when I need to migrate the tests that have liquid staking I'll probably come back to this and rename accordingly.

**Review by bekauz — APPROVED** [review]
got it running on arm64, LGTM!

File: local-interchaintest/src/utils/setup.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (elix1er) [reply local-interchaintest/src/utils/setup.rs:68]: key' the machine, A+

**Comment by elix1er** [comment]
> got it running on arm64, LGTM!

arm64 compability is gold, thx

**Pull request merged**